### PR TITLE
Refactor zookeeper controller

### PR DESCRIPTION
**Pull request by euclidgame**

### CODE REVIEW

File: src/reconciler/exec/io.rs [hunk 1]
```diff
@@ -115,7 +115,7 @@ impl <T: ToView> Response<T> {
 impl <T: ToView> Request<T> {
     pub open spec fn to_view(&self) -> RequestView<T::V> {
         match self {
-            Request::KRequest(req) => RequestView::KRequest(req.to_view()),
+            Request::KRequest(req) => RequestView::KRequest(req@),
             Request::ExternalRequest(req) => RequestView::ExternalRequest(req.to_view()),
```
Comment (marshtompsxd) [inline src/reconciler/exec/io.rs:119]: Why not also replace to_view with view for ExternalRequest?

File: src/controller_examples/zookeeper_controller/common/mod.rs [hunk 2]
```diff
@@ -8,31 +8,60 @@ verus! {
...
-    AfterCreateClientService,
-    AfterUpdateClientService,
-    AfterGetAdminServerService,
-    AfterCreateAdminServerService,
-    AfterUpdateAdminServerService,
-    AfterGetConfigMap,
-    AfterCreateConfigMap,
-    AfterUpdateConfigMap,
+    AfterKRequestStep(ActionKind, SubResource),
+    // AfterGetHeadlessService,
```
Comment (marshtompsxd) [inline src/controller_examples/zookeeper_controller/common/mod.rs:12]: I guess we can remove these commented code now?

File: src/reconciler/exec/io.rs (hunk 1 above)
Comment (euclidgame) [reply src/reconciler/exec/io.rs]: Updated

File: src/controller_examples/zookeeper_controller/common/mod.rs (hunk 2 above)
Comment (euclidgame) [reply src/controller_examples/zookeeper_controller/common/mod.rs]: Updated

**Comment by marshtompsxd** [comment]
@euclidgame I just meged this one https://github.com/vmware-research/verifiable-controllers/pull/348 and can you rebase and push again?
The two lemmas that failed in CI are quite unstable from my experience. We should try our best to make them stable.
Also, besides adding more assertions, pattern matching, and separating into smaller proof functions, you can also try to use `hide` to improve the verification performance. See https://verus-lang.zulipchat.com/#narrow/stream/399078-help/topic/How.20to.20use.20.60hidden.60.20to.20mark.20functions.20as.20opaque

**Comment by marshtompsxd** [comment]
Also I have finished review, you can merge it after you fixed the failed lemmas

**Pull request merged**